packages/backend/test/pg-reset-db-setup.ts

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-console */
+import '../src/config/orm'
+
+import { readdirSync } from 'fs'
+import knex, { Knex } from 'knex'
+import { join } from 'path'
+import { afterEach, beforeEach } from 'vitest'
+
+import config from '../knexfile'
+
+const client = knex(config as Knex.Config)
+
+beforeEach(async () => {
+  // manually running seeds for the same reasons
+  const seedsToRun = readdirSync(config.seeds.directory)
+  for (const seedFile of seedsToRun) {
+    const { seed } = await import(join(config.seeds.directory, seedFile))
+    await seed(client)
+  }
+  console.info(`PostgreSQL seeds run`)
+})
+
+afterEach(async () => {
+  // truncate all tables
+  const tables = await client('pg_catalog.pg_tables')
+    .select('tablename')
+    .where('schemaname', 'public')
+    .whereNotIn('tablename', ['knex_migrations', 'knex_migrations_lock'])
+    .pluck('tablename')
+  for (const table of tables) {
+    await client.raw(`TRUNCATE TABLE "${table}" CASCADE`)
```

Comment:
qq - in theory, if we are truncating all tables, do we need the `CASCADE`? Since everything going to die anyway

Reply:
without CASCADE, we would run into foreign key problems since we might be deleting rows from a table which could referenced from another table